bitbucket-pipelines.yml

```diff
@@ -0,0 +1,91 @@
+          - echo "STARTING THEME CHECK..."
+          - echo "Verifying Ruby, Node.js, and Shopify CLI installations..."
+          - ruby -v      
+          - node -v     
+          - npm -v
+          - shopify version  
+          - ls -al        
+          - echo "Checking if /theme directory exists..."
+          - if [ ! -d "./theme" ]; then echo "/theme directory does not exist."; exit 1; fi
+          - echo "Listing files in the /theme directory..."
+          - find theme -exec ls -ld --time-style=long-iso {} \; | awk '{print $6, $7, $8, $9}'
+          - echo "Running shopify theme check --fail-level=$FAIL_LEVEL --path=theme"
+          - shopify theme check --fail-level=$FAIL_LEVEL --path=theme
+
+    - step: &deploy
+        name: Deploy Theme
+        services:
+          - docker
+        caches:
+          - docker
+        script:
+          - echo "Deploying theme..."
+          - ruby -v
+          - node -v
+          - npm -v
+          - shopify version
+          - ls -al
+          - if [ ! -d "./theme" ]; then echo "/theme directory does not exist."; exit 1; fi
+          - find theme -exec ls -ld --time-style=long-iso {} \; | awk '{print $6, $7, $8, $9}'
+          - |
+            if [ "$BITBUCKET_DEPLOYMENT_ENVIRONMENT" == "staging" ]; then
+              echo "Running shopify theme check --fail-level=$FAIL_LEVEL --path=theme"
+              shopify theme check --fail-level=$FAIL_LEVEL --path=theme
+              echo "Deploying to..."
+              echo -e "  _____ _______       _____ _____ _   _  _____ \n / ____|__   __|/\   / ____|_   _| \ | |/ ____|\n| (___    | |  /  \ | |  __  | | |  \| | |  __ \n \___ \   | | / /\ \| | |_ | | | | .   | | |_ |\n ____) |  | |/ ____ \ |__| |_| |_| |\  | |__| |\n|_____/   |_/_/    \_\_____|_____|_| \_|\_____|\n"
+              echo "shopify theme push --store=$SHOPIFY_STORE --password=$SHOPIFY_ACCESS_TOKEN --theme=$THEME_ID --path=theme --verbose --nodelete"
+              shopify theme push --store="$SHOPIFY_STORE" --password="$SHOPIFY_ACCESS_TOKEN" --theme="$THEME_ID" --path=theme --verbose --nodelete
+            elif [ "$BITBUCKET_DEPLOYMENT_ENVIRONMENT" == "production" ]; then
+              echo "Running shopify theme check --fail-level=$FAIL_LEVEL --path=theme"
+              shopify theme check --fail-level=$FAIL_LEVEL --path=theme
+              echo "Deploying to..."
+              echo -e " _____  _____   ____  _____  _    _  _____ _______ _____ ____  _   _ \n|  __ \|  __ \ / __ \|  __ \| |  | |/ ____|__   __|_   _/ __ \| \ | |\n| |__) | |__) | |  | | |  | | |  | | |       | |    | || |  | |  \| |\n|  ___/|  _  /| |  | | |  | | |  | | |       | |    | || |  | | .   |\n| |    | | \ \| |__| | |__| | |__| | |____   | |   _| || |__| | |\  |\n|_|    |_|  \_|\____/|_____/ \____/ \_____|  |_|  |_____\____/|_| \_|\n"
+              echo "shopify theme push --store=$SHOPIFY_STORE --password=$SHOPIFY_ACCESS_TOKEN --theme=$THEME_ID --path=theme --verbose --allow-live"
```

Comment:
the additional `echo` can be removed because all commands run are available as log outputs in Bitbucket

Reply:
I would prefer to see the actual values used in the commands, as Bitbucket only displays the variable names when executing them.

Example:
![image](https://github.com/user-attachments/assets/80369963-fee0-415f-84f8-98f1b12255c8)

Reply:
this is intentional because repo or deployment variables can be secrets and encrypted. this maybe fine here but i don't think we can use this approach everywhere.